app/components/lock_old_posts.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import datetime as dt
+
+import discord
+
+from app import config
+from app.utils import dynamic_timestamp, post_is_solved
+
+
+async def check_for_old_posts(message: discord.Message) -> None:
+    post = message.channel
+    now = dt.datetime.now(tz=dt.UTC)
+    if (
+        not isinstance(post, discord.Thread)
+        or not post.parent
+        or post.parent.id != config.HELP_CHANNEL_ID
+        or post.locked
+        or post.last_message_id is None
+        or not post_is_solved(post)
+    ):
+        return
+    try:
+        # Ignore messages less than a minute old for the same person may have
+        # sent multiple in quick succession, resulting in their own message
+        # being considered as activity.
+        last = await _get_message(post, 1, before=now - dt.timedelta(minutes=1))
+    except IndexError:
+        # There were no messages older than one minute. Assuming the treshold
+        # below stays at a reasonably large limit, this either means that every
+        # message was deleted, or that there was no response and the original
+        # poster deleted the starter message. It's very unlikely for the post
+        # tono longer be relevant; nor is it likely that anyone would even
+        # message in such a thread. Thus, silently exit if this is the case.
+        return
+    if last.created_at > now - dt.timedelta(seconds=30):
+        return
+
+    try:
+        creation_time_ago = dynamic_timestamp(
+            (await post.fetch_message(post.id)).created_at, "R"
+        )
+    except discord.NotFound:
+        creation_time_ago = "over a month ago"
+    await message.reply(
+        f"This post was created {creation_time_ago} and is likely no longer "
+        "relevant. Please open a new thread instead, making sure to provide "
+        "the required information."
+    )
+    await post.edit(locked=True)
+
+
+async def _get_message(
+    thread: discord.Thread,
+    n: int,
+    *,
+    before: discord.abc.SnowflakeTime | None = None,
+    around: discord.abc.SnowflakeTime | None = None,
+) -> discord.Message:
+    return [
+        message
+        async for message in thread.history(limit=n + 1, before=before, around=around)
+    ][n]
```

Comment:
👇 maybe? So that we don't have to store all messages if we only want the n-th one (even though the only usage is a 2-elem case).
```suggestion
    messages = (
        m async for m in thread.history(limit=n + 1, before=before, around=around)
    )
    return islice(messages, n, None)
```

Reply:
Oh didn't know about `islice`.

Reply:
I forgot to test it before I don't think `islice` works with `AsyncGenerator`?

Reply:
Yeah that's my bad :( There aren't too many built-in async iter utils, so our best bet would probably be to write something like this (since there's no `aenumerate`):
```py
i = 0
async for message in thread.history(...):
    if i == n:
        return message
    i += 1
raise RuntimeError # unreachable or something
```
Or just accept the reality of having a list there 😞

Reply:
> since there's no `aenumerate`

I guess I'll make one then.